spark/src/main/scala/io/delta/sharing/spark/filters/JsonPredicates.scala

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark.filters
+
+import scala.collection.mutable.ListBuffer
+
+import com.fasterxml.jackson.annotation.{JsonSubTypes, JsonTypeInfo}
+
+/**
+ * The evaluation context in which operations will be evaluated.
+ *
+ * @param partitionValues The partition values associated with parquet files.
+ */
+case class EvalContext(partitionValues: Map[String, String])
+
+/**
+ * The data types supported by the filtering operations.
+ */
+object OpDataTypes {
+  val BoolType = "bool"
+  val IntType = "int"
+  val LongType = "long"
+  val StringType = "string"
+  val DateType = "date"
+
+  val supportedTypes = Set(BoolType, IntType, LongType, StringType, DateType)
```

Comment:
How is this list decided?

Reply:
We are starting out with a minimal list (kubit mostly uses date, and I also added basic types).
We will expand this set as needed.